code/SingleRecordAdmin.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace SilverStripe\Admin;
+
+use SilverStripe\Forms\Form;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * The base class for admin sections that exist for editing a single record.
+ */
+abstract class SingleRecordAdmin extends LeftAndMain
+{
+    /**
+     * Determines if there should be a single record in the database that this admin edits.
+     *
+     * If this is not true, you need to provide a mechanism to tell the form which record it should be editing.
+     * This could be an action (e.g. edit/$ID), or could be based on something about the current member, etc.
+     */
+    private static bool $only_one_record = true;
+
+    /**
+     * If no record exists, allow a new one to be created
+     */
+    private static bool $allow_new_record = true;
```

Comment:
For almost all controllers, we'll want to be able to create a record if there isn't one. `CMSProfileController` is the exception. Even with `SiteConfig`, if the record in the DB gets somehow deleted, you'll want to be able to create a new one.

Reply:
Seems like these two new configurable properties shouldn't be configurable? I can't think of a scenario where they would be better of simply being protected non-static properties instead? There's a miniscule performance benefit, though more importantly it prevents people from doing things they shouldn't via config, like changing the the config values of SecurityAdmin

Reply:
The pattern in these controllers is to use configuration for this sort of thing, so I was just following that pattern.
If you feel strongly about it, I can change them to protected member properties.

Reply:
Yup, I meant CMSProfileController

You're probably right it is something of a pattern, which it probably shouldn't be though we should respect what's already there.